deploy-agent/tests/unit/deploy/common/test_stats.py

```diff
@@ -119,13 +126,23 @@ def test_deserialize(self):
         self.assertFalse(stat.deserialize(ins=invalid_type))
 
 
+class TestMetricClientExceptions(unittest.TestCase):
+    def test__MetricClientConfigurationError(self):
+        with self.assertRaises(MetricClientConfigurationError):
+            MetricClient(port=None, cache_path=None)
+
+
 class TestMetricClient(unittest.TestCase):
     cache_path = 'tests/unit/deploy/common/test_stats.cache'
```

Comment:
https://docs.python.org/3/library/tempfile.html may be useful

Reply:
although a good suggestion skipping here to match expected inputs that this test case covers